site/_plugins/canonical_url.rb

```diff
@@ -0,0 +1,20 @@
+# Jekyll::Page#url is memoized on first call, but this causes issues with the
+#   paginator.  As such, we get the URL and then remove the memoization.
+def extract_page_url(page)
+  had_url = true if page.instance_variable_defined?(:@url)
```

Comment:
why not the simpler `had_url = page.instance_variable_defined?(:@url)` ?

Reply:
Oh yeah, that's better...I had that originally, but changed things, and somehow ended with this.